src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -1,78 +1,109 @@
+                 blob_sas_url):
+        match = _get_match(blob_sas_url)
+        account_name = match.group('account_name')
+        endpoint_suffix = match.group('endpoint_suffix')
+        container_name = match.group('container_name')
+        blob_name = match.group('blob_name')
+        sas_token = match.group('sas_token')
+
+        if not account_name or not container_name or not blob_name or not sas_token:
+            raise CLIError("Failed to parse the sas url: '{!s}'."
+                           .format(blob_sas_url))
+
+        self.account_name = account_name
+        self.endpoint_suffix = endpoint_suffix
+        self.container_name = container_name
+        self.blob_name = blob_name
+        self.sas_token = sas_token
 
 
 def acr_build_show_logs(cmd,
                   client,
                   registry_name,
                   build_id,
                   resource_group_name=None):
+
     resource_group_name = get_resource_group_name_by_registry_name(
         cmd.cli_ctx, registry_name, resource_group_name)
 
     # Required and hardcoded for now. Refer to the "HACK" introduced in build_operations.py
     custom_headers = dict()
     custom_headers['logType'] = 'RawText'
     build_log_result = client.get_log_link(
-        build_id=build_id, resource_group_name=resource_group_name,
+        build_id=build_id, resource_group_name=resource_group_name, 
         registry_name=registry_name, custom_headers=custom_headers)
     log_file_sas = build_log_result.log_link
 
     if not log_file_sas:
         return 'No logs found.'
 
-    match = __get_match(log_file_sas)
-    account_name = match.group('account_name')
-    container_name = match.group('container_name')
-    blob_name = match.group('blob_name')
-    sas_token = match.group('sas_token')
-
-    if not account_name or not container_name or not blob_name or not sas_token:
-        raise CLIError("Failed to parse the log URL: '{!s}'."
-                       .format(log_file_sas))
-
-    blob_service = AppendBlobService(account_name=account_name,
-                                     sas_token=sas_token)
+    blob_info = BlobInfo(log_file_sas)
 
     byte_size = 1024*4
     timeout_in_minutes = 10
     timeout_in_seconds = timeout_in_minutes * 60
-    now = time.time()
-    __stream_logs(byte_size, timeout_in_seconds, blob_service,
-                container_name, blob_name)
+
+    _stream_logs(byte_size, timeout_in_seconds, 
+                AppendBlobService(account_name=blob_info.account_name, sas_token=blob_info.sas_token, endpoint_suffix=blob_info.endpoint_suffix),
+                blob_info.container_name, blob_info.blob_name)
 
 
-def __get_match(sas_url):
+def _get_match(sasURL):
     return re.search(
-        (r"http(s)?://(?P<account_name>.*?)\..*?/(?P<container_name>.*?)/"
-            r"(?P<blob_name>.*?)\?(?P<sas_token>.*)"), sas_url)
+        (r"http(s)?://(?P<account_name>.*?)\.blob\.(?P<endpoint_suffix>.*?)/(?P<container_name>.*?)/"
```

Comment:
Just curious: when's the case where `endpoint_suffix` is required (it defaults to `core.windows.net`)?